src/Kuzzle.ts

```diff
@@ -815,4 +846,26 @@ Discarded request: ${JSON.stringify(request)}`));
 
     dequeuingProcess();
   }
+
+  /**
+   * Sends a request with a timeout
+   */
+  private _timeoutRequest(delay, {request, options = {}}) {
+    // No timeout
+    if (delay === -1) {
+      return this.protocol.query(request, options);
+    }
+
+    const timeout = new Promise((resolve, reject) => {
+      const id = setTimeout(() => {
+        clearTimeout(id);
+        reject(`Request timed out after ${delay} ms`);
+      }, delay);
+    });
+
+    return Promise.race([
+      timeout,
+      this.protocol.query(request, options)
```

Comment:
Open debate / reflection:
Doing this will not cancel the original request, it may still be executed after a certain amount of time.
I don't think we can do something about it actually but at least we should add a warning in the documentation

Reply:
That's right, the only thing this will do is prevent the SDK from being frozen if a request takes too long to be resolved, but Kuzzle will still resolve it at some point. If we really want to stop the request from being process after an amount of time, we should do it directly in Kuzzle and not in the SDK.

That said I don't know if that's what the issue was for

Any ideas ?

Reply:
Yes you are definitely right :+1: But even on Kuzzle side I don't think we can do much since we cannot remove a promise from the even loop.

I just thing we should add a warning in the documentation telling that requests are errored on the SDK side after the timeout but they still may be processed by Kuzzle